File: app/src/main/java/com/sasfmlzr/findfilm/request/FindFilmApi.java
package com.sasfmlzr.findfilm.request;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface FindFilmApi {
    @GET("https://api.themoviedb.org/3/discover/movie")
    Call<DiscoverMovieRequest> getDiscoverMovie(@Query("api_key") String API_KEY,
                                                @Query("language") String language,
                                                @Query("page") int page);

    @GET("https://api.themoviedb.org/3/search/movie")
    Call<DiscoverMovieRequest> getSearchMovie(@Query("api_key") String API_KEY,
                                              @Query("language") String language,
                                              @Query("query") String movie,
                                              @Query("page") int page);

    @GET("https://api.themoviedb.org/3/movie/{movie_id}")
    Call<CurrentMovieRequest> getCurrentMovie(@Path("movie_id") int movie_id,
                                              @Query("api_key") String API_KEY,
                                              @Query("language") String language);
}

File: app/src/main/java/com/sasfmlzr/findfilm/request/FindFilmApi.kt
package com.sasfmlzr.findfilm.request

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface FindFilmApi {
    @GET("https://api.themoviedb.org/3/discover/movie")
    fun getDiscoverMovie(
        @Query("api_key") API_KEY: String?,
        @Query("language") language: String?,
        @Query("page") page: Int
    ): Call<DiscoverMovieRequest>

    @GET("https://api.themoviedb.org/3/search/movie")
    fun getSearchMovie(
        @Query("api_key") API_KEY: String?,
        @Query("language") language: String?,
        @Query("query") movie: String?,
        @Query("page") page: Int
    ): Call<DiscoverMovieRequest>

    @GET("https://api.themoviedb.org/3/movie/{movie_id}")
    fun getCurrentMovie(
        @Path("movie_id") movie_id: Int,
        @Query("api_key") API_KEY: String?,
        @Query("language") language: String?
    ): Call<CurrentMovieRequest>
}